randovania/games/dread/json_data/Dairon.txt

```diff
@@ -2405,38 +2368,69 @@ Extra - asset_id: collision_camera_025
   > Door to Map Station
       All of the following:
           Any of the following:
-              Gravity Suit
+              Gravity Suit or Before Dairon - powergenerator_002
               Heat/Cold Runs (Intermediate) and Cold Damage ≥ 150
           Space Jump or Speed Booster or Walljump (Beginner) or Simple IBJ
   > Door to Power Switch 2
       Any of the following:
-          # TODO: Implement logic for un-frozen freezer, which adds a few extra requirements for jumping around the pond
-          Gravity Suit
-          Heat/Cold Runs (Intermediate) and Cold Damage ≥ 200
-  > Door to Energy Recharge Station West (doorpowerpower_030)
+          All of the following:
+              After Dairon - powergenerator_002
+              Any of the following:
+                  Gravity Suit
+                  Heat/Cold Runs (Intermediate) and Cold Damage ≥ 200
+          All of the following:
+              Before Dairon - powergenerator_002
+              Any of the following:
+                  Grapple Beam or Spider Magnet or Space Jump or Speed Booster
+                  All of the following:
+                      Gravity Suit
+                      Spin Boost or Walljump (Beginner) or Simple IBJ
+                  Flash Shift and Walljump (Beginner)
+  > Door to Energy Recharge Station West (Upper)
+      All of the following:
+          Speed Booster
```

Comment:
Currently, "Door to Energy Recharge Station West (Upper)" -> "Freezer Center" doesn't have a speedboost requirement, and if a room shuffle were to occur, then the speedboost blocks would block the player. However, I don't think it's possible to charge a speedboost from the top here.

It's finefor now, but I'm not sure what the best long-term solution is - maybe an event for destroying those speedboost blocks? At the very minimum, I would add a "TODO" comment saying that this eventually needs to be addressed.

Reply:
I considered adding an event, but then remembered you can't even get up there without destoyring the speed blocks anyway. And IIRC, room rando is never going to be a thing. That would be a thing to ask Dunc probably.